content/docs/operating/configuration.md

```diff
@@ -456,6 +460,50 @@ names:
 Where `<filename_pattern>` may be a path ending in `.json`, `.yml` or `.yaml`. The last path segment
 may contain a single `*` that matches any character sequence, e.g. `my/path/tg_*.json`.
 
+### EC2 SD configurations `<ec2_sd_config>`
+
+CAUTION: EC2 SD is in beta: breaking changes to configuration are still
+likely in future releases.
+
+EC2 SD configurations allow retrieving scrape targets from AWS EC2 instances. The private IP address is used by default, but my be changed to the public IP address with relabeling.
+
+The following meta labels are available on targets during relabeling:
+
+* `__meta_ec2_instance_id`: the EC2 instance ID
+* `__meta_ec2_public_ip`: the public IP address of the instance
+* `__meta_ec2_private_ip`: the private IP address of the instance, if present
+* `__meta_ec2_tag_<tagkey>`: each tag value of the instance
+
+See below for the configuration options for EC2 discovery:
+
+```
+# The information to access the EC2 API.
+
+# The AWS Region.
+region: <string>
+
+# The AWS API keys. If blank, the environment variables `AWS_ACCESS_KEY_ID`
+# and `AWS_SECRET_ACCESS_KEY` are used.
+[ access_key: <string> ]
+[ secret_key: <string> ]
+
+# Refresh interval to re-read the instance list.
+[ refresh_interval: <duration> | default = 60s ]
+
+# The port to scrape metrics from. If using the public IP address, this must
+# instead be specified in the relabeling rule.
+[ port: <int> | default = 80 ]
+```
+
+For example, to use the public IP address on port 8080, the following relabeling may be done:
```

Comment:
This is reference documentation, so we should avoid examples here to keep it brief.